plaso/engine/artifact_filters.py

```diff
@@ -0,0 +1,239 @@
+                            'HKEY_LOCAL_MACHINE\\SOFTWARE',
+                            'HKEY_LOCAL_MACHINE\\SAM',
+                            'HKEY_LOCAL_MACHINE\\SECURITY']
+STANDARD_OS_FILTERS = ['windows',
+                       'linux',
+                       'darwin']
+RECURSIVE_GLOB_LIMIT = 10
+
+
+class ArtifactFilters(object):
+  """Forensic artifacts filters.
+
+  Build extraction and parsing filters from forensic artifact definitions.
+
+  Forensic artifacts are defined in:
+  https://github.com/ForensicArtifacts/artifacts/blob/master/docs/Artifacts%20definition%20format%20and%20style%20guide.asciidoc
+  """
+
+  def __init__(self, artifacts_registry, filters, knowledge_base):
+    """Initializes a filter file.
+
+    Args:
+      path (str): path to a file that contains one or more forensic artifacts.
+      knowledge_base (KnowledgeBase): Knowledge base for Log2Timeline.
+    """
+    super(ArtifactFilters, self).__init__()
+    self._artifacts_registry = artifacts_registry
+    self._filters = filters
+    self._knowledge_base = knowledge_base
+
+  def BuildFindSpecs(self, environment_variables=None):
+    """Build find specification from a forensic artifacts file.
+
+    Args:
+      environment_variables (Optional[list[EnvironmentVariableArtifact]]):
+          environment variables.
+    """
+    find_specs = {}
+
+    filtered_defintions = []
+    # Check length and values
+    if len(self._filters) == 1 and self._filters[0] in STANDARD_OS_FILTERS:
+      for definition in self._artifacts_registry.GetDefinitions():
+        if self._filters[0].lower() in (
+            os.lower() for os in definition.supported_os):
+          filtered_defintions.append(definition)
+    else:
+      for artifact_filter in self._filters:
+        if self._artifacts_registry.GetDefinitionByName(artifact_filter):
+          filtered_defintions.append(
+            self._artifacts_registry.GetDefinitionByName(artifact_filter))
+
+    for definition in filtered_defintions:
+      for source in definition.sources:
+        if source.type_indicator == artifact_types.TYPE_INDICATOR_FILE:
+          for path_entry in source.paths:
+            self.BuildFindSpecsFromFileArtifact(
+              path_entry, source.separator, environment_variables, find_specs)
+        elif (source.type_indicator ==
+              artifact_types.TYPE_INDICATOR_WINDOWS_REGISTRY_KEY):
+          keys = set(source.keys)
+          for key_entry in keys:
+            if self._CheckKeyCompatibility(key_entry):
+              self.BuildFindSpecsFromRegistryArtifact(
+                key_entry, find_specs)
+        elif (source.type_indicator ==
+              artifact_types.TYPE_INDICATOR_WINDOWS_REGISTRY_VALUE):
+          # TODO: Handle Registry Values Once Supported in dfwinreg.
+          logging.warning(('Unable to handle Registry Value, extracting '
+                           'key only: {0:s} ').format(source.key_value_pairs))
```

Comment:
source.key_value_pairs is a list of tuples, so {:s} is likely the wrong formatter

use an explicit join e.g. ','.join(['(key: {0:s}, value: {1:s})'.format(key_path, value_name) for key_path, value_name in source.key_value_pairs])

Reply:
Merged in #1883